File: src/game/controller/Command.java
package game.controller;

import game.models.Direction;

public enum Command {
    MOVE_UP,
    MOVE_DOWN,
    MOVE_RIGHT,
    MOVE_LEFT,
    PLACE_BOMB,
    PAUSE,
    INTERACT;

    public Direction commandToDirection(){
        switch (this){
            case MOVE_UP: return Direction.UP;
            case MOVE_DOWN: return Direction.DOWN;
            case MOVE_LEFT: return Direction.LEFT;
            case MOVE_RIGHT: return Direction.RIGHT;
        }
        return null;
    }
}

File: src/game/input/game/Command.kt
package game.input.game

import game.domain.world.domain.entity.geo.Direction

enum class Command {
    MOVE_UP, MOVE_DOWN, MOVE_RIGHT, MOVE_LEFT, ATTACK, PAUSE, INTERACT;

    fun commandToDirection(): Direction? {
        return when (this) {
            MOVE_UP -> Direction.UP
            MOVE_DOWN -> Direction.DOWN
            MOVE_LEFT -> Direction.LEFT
            MOVE_RIGHT -> Direction.RIGHT
            else -> null
        }
    }
    fun isMovementKey(): Boolean {
        return arrayOf(MOVE_UP, MOVE_DOWN, MOVE_RIGHT, MOVE_LEFT).contains(this)
    }
}